subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBase.java

```diff
@@ -686,26 +683,18 @@ public List<SubscriptionBillingEvent> getSubscriptionBillingEvents(final Version
 
                     if (isCreateOrTransfer || isChangeEvent || isPhaseEvent) {
 
-                        // We are moving to a new Plan, we use the latest active catalog version at the time this operation took place.
-                        final StaticCatalog catalogVersion = catalog.versionForDate(billingTransition.getEffectiveDate());
-
-                        final Plan currentPlan;
-                        try {
-                            currentPlan = catalogVersion.findPlan(billingTransition.getPlan().getName());
-                        } catch (final CatalogApiException e) {
-                            if (e.getCode() == ErrorCode.CAT_NO_SUCH_PLAN.getCode()) {
-                                // Retired plan
-                                continue;
-                            } else {
-                                throw e;
-                            }
-                        }
+                        // If we are moving to a new Plan, we use the latest active catalog version at the time this operation took place.
+                        final DateTime billingTransitionEffectiveDate = isPhaseEvent ? lastPlanTransition.getEffectiveTransitionTime() : billingTransition.getEffectiveDate();
```

Comment:
Is this required? Or could we just set `billingTransitionEffectiveDate=billingTransition.getEffectiveDate()` every time? Not saying this is wrong, just trying to understand the code better  ;-)

Reply:
It is not easy to understand, I added this fix after I discovered that one of our test, i.e `TestCatalogNewEntries#testNewPlanWithRetiredOldPlan` was failing:

Given 2 catalog versions, V1 effDt = `2011-01-01` and V2 `effDt 2015-12-01`: V1 has a plan `pistol-monthly` with a 30 days TRIAL  but such plan got deprecated in V2.

If we start a subscription on `2015-11-01` with plan `pistol-monthly`: The code above processes the PHASE event on `2015-12-01` (`2015-11-01` + `30` days), and unless we look for the Plan in the original catalog version V1, it will fail because such plan got deprecated in V2, hence the computation of the `billingTransitionEffectiveDate`.